src/HackF5.UnitySpy/Detail/ProcessFacade.cs

```diff
@@ -229,14 +229,16 @@ private object[] ReadManagedArray(TypeInfo type, IntPtr address)
             var arrayDefinition = type.Image.GetTypeDefinition(arrayDefinitionPtr);
             var elementDefinition = type.Image.GetTypeDefinition(this.ReadPtr(arrayDefinitionPtr));
 
+            // Not sure why in the new version of unity the array it seems to have pointers instead of the element
+            // Maybe mono changed?
             var elementSize = MonoLibraryOffsets.UsesArrayDefinitionSize ? arrayDefinition.Size : SizeOfPtr;
 
             var count = this.ReadInt32(ptr + SizeOfPtr * 3);
-            var start = ptr + SizeOfPtr * 4;
+            var start = ptr + (SizeOfPtr * 4);
             var result = new object[count];
             for (var i = 0; i < count; i++)
             {
```

Comment:
Not sure why this 3 and 4 values are here, I just took the original fixed ones (0xc and 0x10) and replaced with SizeOfPtr offsets to ensure 64bits and 32bits support